### PR TITLE
[IMPORTANT] READ THE FOLLOWING CAREFULLY

### PR DESCRIPTION
**Pull request by ileuxali**
Refactored data access, initialized use case, made new package for classes might/will be used in multiple use cases
- Please read the javadoc for the new data access class SerializeDataAccess.
   - The UpdateUserDSGateway is an example empty DSGateway, it will be deleted on a future pull request, everyone needs their own DSGateway for individual use cases
- Ignore the files related to my use case under the package with use case name (still working on them).
- Made a new package for classes that might/will be used in multiple usecases.
- If you have any questions please message me either here as a comment or on instagram 
- :^)

### CODE REVIEW

**Review by SharonHsieh22 — APPROVED** [review]
Thanks for making the changes we discussed and implementing the data access class. Everything looks good!

**Review by aaryavatsa — APPROVED** [review]
Everything looks good, great work on SerializeDataAccess !

File: src/main/java/MultiUsecaseUtil/SerializeDataAccess.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            output.close();
+            buffer.close();
+            file.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public HashMap<String, User> loadUsers(){
```
Comment (helenaglow) [inline src/main/java/MultiUsecaseUtil/SerializeDataAccess.java:139]: To get the current User by username, would we loadUsers() and then search for the user in the Hashmap? Or are we still going to implement a loadUser(String username)?

Comment (ileuxali) [reply src/main/java/MultiUsecaseUtil/SerializeDataAccess.java:139]: implement a loaduser in your dsgateway interface

**Review by helenaglow — APPROVED** [review]
Everything looks good. Thank you for the really in depth descriptions. I will adjust my dsgateway to align with your SerializeDataAccess

**Pull request merged**